src/libraries/Microsoft.PowerFx.Core/Entities/External/IExternalOptionSet.cs

```diff
@@ -20,8 +20,77 @@ internal interface IExternalOptionSet : IExternalEntity
         /// </summary>
         IEnumerable<DName> OptionNames { get; }
 
+        /// <summary>
+        /// Backing kind for this option set. Must be Number, String, Boolean, or Color.
```

Comment:
>Must be Number, String, Boolean, or Color. [](http://example.com/codeflow?start=46&length=42)

We should have checks that enforce that (possibly when adding a new option set to the symbol table)

Reply:
I don't know of a way to enforce this at the interface level.  Having a seperate OptionSetDKind is possible, but gets messy considering DKind is a part of DType.  This is enforced in EnumSymbol through a Contracts.Assert.  OptionSet.cs only supports creating String backed option sets.